src/duration.rs

```diff
@@ -420,6 +434,67 @@ impl Duration {
         })
     }
 
+    fn parse_oversize_hour_time_format(bytes: &[u8], offset: usize) -> Result<Duration, ParseError> {
+        // handle format H..H:MM:DD. So we already know the fix format for the last 6 chars and
+        // minimum length is 8;
+        let byte_len = bytes.len();
+        if byte_len < 8 {
+            return Err(ParseError::TooShort);
+        }
+
+        let hour_numeric_limit = 24 * (1e9 as i64);
+
+        // see if the ":" is where it should be in "H..H:MM:DD".
+        if bytes.get(byte_len - 3) != Some(&b':') || bytes.get(byte_len - 6) != Some(&b':') {
+            return Err(ParseError::DurationInvalidFraction);
+        }
+
+        let mut hour_numeric_value: i64 = 0;
+        let hour_byte_len = byte_len - 6;
+        // the limit for H..H is 1,000,000,000 * 24 => result have length of 11 chars.
+        // so just checking if it can fail early.
+        if hour_byte_len > 11 || (hour_byte_len == 11 && (bytes.get(offset) > Some(&b'2'))) {
+            return Err(ParseError::DurationHourValueTooLarge);
+        }
+
+        // parsing the H..H in to a number.
+        for idx in offset..hour_byte_len {
+            let n = bytes.get(idx).ok_or(ParseError::InvalidCharHour)? - b'0';
+            if n > 9 {
+                return Err(ParseError::InvalidCharHour);
+            }
+            hour_numeric_value = (hour_numeric_value * 10) + (n as i64);
+        }
+
+        if hour_numeric_value >= hour_numeric_limit {
+            return Err(ParseError::DurationHourValueTooLarge);
+        }
+
+        // extract the Day value and parse the left over time part.
+        let days = (hour_numeric_value / 24) as u32;
+        let leftover_hour = (hour_numeric_value % 24) as i32;
+
+        let h1 = (leftover_hour / 10) as u8 + b'0';
+        let h2 = (leftover_hour % 10) as u8 + b'0';
+
+        let mut temp_vec: Vec<u8> = vec![h1, h2];
+        temp_vec.extend_from_slice(&bytes[byte_len - 6..]);
```

Comment:
Rather than calculating remainder hours and reusing all of `parse_time`, should we make a `parse_minutes_seconds`?

Reply:
I sure can, didn't think of that approach at the time. Thank @davidhewitt